transmart-rest-api/src/main/groovy/org/transmartproject/rest/dataExport/ExportJobExecutor.groovy

```diff
@@ -14,49 +16,45 @@ import java.util.zip.ZipOutputStream
  * This class will encompass the job scheduled by Quartz.
  */
 @Slf4j
-class ExportJobExecutor implements Job {
-
-    def ctx = Holders.grailsApplication.mainContext
+class ExportJobExecutor implements InterruptableJob {
 
+    ApplicationContext ctx = Holders.applicationContext
     ExportService exportService = ctx.restExportService
     ExportAsyncJobService asyncJobService = ctx.exportAsyncJobService
 
+    ZipOutputStream zipFile
+
+    @Override
     void execute(JobExecutionContext jobExecutionContext) {
         Map jobDataMap = jobExecutionContext.jobDetail.getJobDataMap()
 
-        PersistenceContextInterceptor interceptor
+        PersistenceContextInterceptor interceptor = ctx.persistenceInterceptor
         try {
-            interceptor = Holders.applicationContext.persistenceInterceptor
             interceptor.init()
-            zipData(jobDataMap)
-        } catch (UnexpectedResultException e) {
```

Comment:
I agree that it's much cleaner like this, but now too many possible cases are missed this way. Consider the case that some filesystem error occurs when accessing the working directory. Also, please change `Exception` to `Throwable`, now assertion errors are not caught).